synapse/appservice/api.py

```diff
@@ -103,16 +106,22 @@ def query_alias(self, service, alias):
     @defer.inlineCallbacks
     def query_3pe(self, service, kind, protocol, fields):
         if kind == ThirdPartyEntityKind.USER:
-            uri = "%s/thirdparty/user/%s" % (service.url, urllib.quote(protocol))
+            fragment = "user"
             required_field = "userid"
         elif kind == ThirdPartyEntityKind.LOCATION:
-            uri = "%s/thirdparty/location/%s" % (service.url, urllib.quote(protocol))
+            fragment = "location"
```

Comment:
A good point. We might as well move the constants into `api.constants` and rely on values directly.

Done 9459137